conftest.py

```diff
@@ -17,7 +25,45 @@ def staffuser():
 
 
 @pytest.yield_fixture()
-def app(request):
+def storage_dir(monkeypatch, tmpdir, settings):
+    """Return a new temporary storage dir.
+    Settings are also modified to point to the directory."""
+    STORAGE_ROOT = settings.STORAGE_ROOT
+
+    # Get the paths to changes and their original values.
+    original_paths = (k for k in dir(settings.wrapped_object)
+                      if k.isupper() and not k.startswith('__'))
+    original_paths = (k for k in original_paths
+                      if k.endswith('_ROOT') or k.endswith('_DIR'))
+    original_paths = {k: getattr(settings, k) for k in original_paths}
+    original_paths = {k: v for k, v in original_paths.items() if v}
+
+    # Update paths replacing STORAGE_ROOT by tmpdir.
+    for k, v in original_paths.items():
+        rel_path = os.path.relpath(v, STORAGE_ROOT)
+        new_temp_path = os.path.join(str(tmpdir), rel_path)
+        try:
+            os.makedirs(new_temp_path)
```

Comment:
Maybe add an `exist_ok=True`, just to be safe? (in case two settings use the same value)